app/src/main/java/edu/uwstout/p2pchat/HomeFragment.java

```diff
@@ -249,16 +257,25 @@ public void peerConnectionSucceeded(WifiP2pDevice device)
     {
         Toast.makeText(this.getContext(), "Connection succeeded", Toast.LENGTH_LONG).show();
         ViewModel viewModel = new ViewModel(getActivity().getApplication());
-        if (!viewModel.peerExists(device.deviceAddress))
-        {
-            viewModel.insertPeer(device.deviceAddress, "nickname");
-        }
-        HomeFragmentDirections.ChatAction action = HomeFragmentDirections.chatAction();
 
-        action.setAddress(device.deviceAddress);
+       liveData.observeForever(new Observer<List<Peer>>()
+       {
+           @Override
+           public void onChanged(List<Peer> peers)
+           {
+              if(!peerExists(peers,device.deviceAddress))
+              {
+                  viewModel.insertPeer(device.deviceAddress,"Nick");
```

Comment:
Why does this contain a hard-coded "Nick"?

Reply:
See issue 87. This issue requires 87